src/FsCheck/Testable.fs

```diff
@@ -103,35 +93,51 @@ module private Testable =
             fun _  -> Gen.constant t
             |> Property
 
-        let ofResult (r:ResultContainer) : Property = 
+        let ofResult (r:Result) : Property = 
             r
             |> Shrink.ofValue
             |> ofShrinkResult
          
         let ofBool b =
             Res.ofBool b
-            |> Value
             |> ofResult
         
         let ofTaskBool (b:Task<bool>) :Property = 
-            b.ContinueWith (fun (x:Task<bool>) -> 
-                match (x.IsCanceled, x.IsFaulted) with
-                    | (false,false) -> Res.ofBool x.Result
-                    | (_,true) -> Res.failedException x.Exception
-                    | (true,_) -> Res.failedCancelled)
-            |> Future
+            try 
+                b.Wait()
+            with
+                | :? AggregateException -> ()
+
+            match (b.IsCanceled, b.IsFaulted) with
+                | (false,false) -> Res.ofBool b.Result
+                | (_,true) -> Res.failedException b.Exception
+                | (true,_) -> Res.failedCancelled
             |> ofResult
+
+        let ofTaskProperty (b:Task<Property>) :Property = 
+            try 
+                b.Wait()
```

Comment:
> Why not remove `ResultContainer` altogether and force Task result on the level where Property is created out of Testable?
> Do you see any issues with the approach?

I think this call to `Task.Wait()` would be problematic.

Reply:
Yes agreed - it looks like this change would (at least partly) break parallel and/or async execution, Such changes are unfortunately hard to test, but the original contribution of parallel testing did verify actual speedups on non-trivial tests.